src/routes/admin/frontdesk/Cart.svelte

```diff
@@ -1,19 +1,14 @@
 <script lang="ts">
-  export let selectedItem;
+  let { selectedItem } = $props();
 
-  // FIXME when upgrade to svelte 5:
-  // https://svelte.dev/tutorial/svelte/deep-state
-  // let cart = $state([]);
-  let cart = [];
-  $: totalPrice = cart.reduce((sum, item) => sum + item.price, 0);
+  let cart = $state([]);
+  let totalPrice = $derived(cart.reduce((sum, item) => sum + item.price, 0));
   function addToCart() {
-    // cart.append(selectedItem);
-    cart = [...cart, selectedItem];
+    cart.push(selectedItem);
   }
 
   function removeFromCart(index) {
-    // cart.splice(index, 1);
-    cart = cart.slice(0, index).concat(cart.slice(index + 1));
+    cart = cart.filter((_, i) => i !== index);
```

Comment:
Hvorfor ikke bruke splice her?

Reply:
Helt subjektivt, og ment som et forslag. Er du uenig?

Reply:
Hadde for øvrig vært noe å tenke på å bygge opp funksjoner i /utils

For eksmepel her, er jo det en `removeAt(array, index)` uansett hvilken implementasjon som brukes.

Reply:
> Hadde for øvrig vært noe å tenke på å bygge opp funksjoner i /utils
> 
> For eksmepel her, er jo det en `removeAt(array, index)` uansett hvilken implementasjon som brukes.

Splice er innebygget i JS, og gjør akkurat dette. det er ikke noe poeng i å overkomplisere det noe mer enn det. Som med `.push()` i den andre funksjonen, kan vi bare mutere listen på samme måte heller enn å reassigne den, som er grunnen til at vi trengte deep state fra svelte 5.

Reply:
Jajaja, stemmer! Mulig man kan bare kjøre en delete på verdien.

Splice splicer jo da, den gjør ikke det samme som removeAt